### PR TITLE
Add rnpm-plugin-windows and pom.xml check

### PR DESCRIPTION
**Pull request by bowdenk7**
add windows react native npm package to track for mobile extension team and pom.xml for java extension team

also added `devDependencies` as a check location for npm packages because `rnpm-plugin-windows` is typically installed there.

### CODE REVIEW

File: src/vs/workbench/contrib/stats/node/workspaceStats.ts [hunk 1]
```diff
@@ -502,18 +507,18 @@ export class WorkspaceStats implements IWorkbenchContribution {
 			const packageJsonPromises = getFilePromises('package.json', this.fileService, this.textFileService, content => {
 				try {
 					const packageJsonContents = JSON.parse(content.value);
-					if (packageJsonContents['dependencies']) {
+					if (packageJsonContents['dependencies'] || packageJsonContents['devDependencies']) {
 						for (let module of ModulesToLookFor) {
 							if ('react-native' === module) {
-								if (packageJsonContents['dependencies'][module]) {
+								if (packageJsonContents['dependencies'][module] || packageJsonContents['devDependencies'][module]) {
```
Comment (sbatten) [inline src/vs/workbench/contrib/stats/node/workspaceStats.ts]: dependencies and devDependencies are now possibly undefined

File: src/vs/workbench/contrib/stats/node/workspaceStats.ts [hunk 2]
```diff
@@ -502,18 +507,18 @@ export class WorkspaceStats implements IWorkbenchContribution {
...
 									tags['workspace.reactNative'] = true;
 								}
 							} else if ('tns-core-modules' === module) {
-								if (packageJsonContents['dependencies'][module]) {
+								if (packageJsonContents['dependencies'][module] || packageJsonContents['devDependencies'][module]) {
```
Comment (sbatten) [inline src/vs/workbench/contrib/stats/node/workspaceStats.ts]: same for this and below

**Review by sbatten — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/stats/node/workspaceStats.ts (hunk 1 above)
Comment (bowdenk7) [reply src/vs/workbench/contrib/stats/node/workspaceStats.ts]: I think I addressed this, let me know if there is a better/cleaner way to do the `undefined` check.

Comment (sbatten) [reply src/vs/workbench/contrib/stats/node/workspaceStats.ts]: Looks right now. Did you validate the events already?

**Review by sbatten — APPROVED** [review]

**Review by sbatten — APPROVED** [review]

File: src/vs/workbench/contrib/stats/node/workspaceStats.ts (hunk 1 above)
Comment (bowdenk7) [reply src/vs/workbench/contrib/stats/node/workspaceStats.ts]: yep - just need to make sure the GDPR tool gets ran.

**Pull request merged**